### PR TITLE
:bug: fix custom reports crashing when opening table

### PR DESCRIPTION
**Pull request by MatissJanis**
State mutations should not be happening during render cycle.

Reproduction:
1. create custom report (type: table)
2. save it
3. open the main reports page
4. click on the saved custom report
5. 💥

### CODE REVIEW

File: packages/desktop-client/src/components/reports/reports/CustomReport.tsx [hunk 1]
```diff
@@ -387,6 +387,7 @@ export function CustomReport() {
   ]);
 
   const getGraphData = useMemo(() => {
+    // TODO: fix me - state mutations should not happen inside `useMemo`
```
Comment (carkom) [inline packages/desktop-client/src/components/reports/reports/CustomReport.tsx:390]: Is this something you want to fix in this PR or is it being flagged to be completed in a different PR?

**Comment by carkom** [comment]
Seems like a fairly major bug, I'm surprised you're the first one to point it out. Thanks for finding it and calling it out.

File: packages/desktop-client/src/components/reports/reports/CustomReport.tsx (hunk 1 above)
Comment (MatissJanis) [reply packages/desktop-client/src/components/reports/reports/CustomReport.tsx:390]: That's for a different PR. I _might_ pick it up when integrating dashboards more tightly with custom reports. But I might also not. Either way - still good to call it out so eventually it's patched.

**Review by MikesGlitch — APPROVED** [review]

**Review by carkom — APPROVED** [review]
LGTM
